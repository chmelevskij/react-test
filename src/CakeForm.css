.CakeForm {
  display: flex;
  flex-direction: column;
  background: rgba(255, 255, 255, 0.5);
  border-radius: 5px;
  color: black;
  padding: 10px;
}

.CakeForm-submit {
  line-height: 2rem;
  background: #9c92ac;
}

input,
textarea {
  margin: 10px;
  padding: 5px;
  border: none;
  border-radius: 5px;
}
