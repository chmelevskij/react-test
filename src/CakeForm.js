import React from 'react';
import './CakeForm.css'

const CakeForm = ({handleInput, title, desc, image, onSubmit}) => (
  <form className="CakeForm">
    <label htmlFor="title"> Title </label>
    <input className="CakeForm-input" id="title" value={title} onChange={handleInput} name="title" type="text" placeholder="Title" required/>
    <label htmlFor="desc"> Description </label>
    <textarea  className="CakeForm-input" id="desc" value={desc} name="desc" onChange={handleInput} placeholder="Description" cols="30" rows="10"></textarea>
    <label htmlFor="image"> Image </label>
    <input className="CakeForm-input" id="image" value={image} onChange={handleInput} name="image" type="url" placeholder="Image URL"/>
    <button className="CakeForm-submit CakeForm-input" type="submit" onClick={onSubmit}>Add</button>
  </form>
)

export default CakeForm
