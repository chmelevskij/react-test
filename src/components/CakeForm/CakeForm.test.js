import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CakeForm from './CakeForm.js'

describe('<CakeForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<CakeForm />, div)
  })

  it('Renders cakes', () => {
    let wrapper = renderer.create(<CakeForm />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
