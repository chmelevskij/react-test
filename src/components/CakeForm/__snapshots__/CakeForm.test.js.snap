// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CakeForm /> Renders cakes 1`] = `
<form
  className="CakeForm"
  style={
    Object {
      "display": "none",
    }
  }
>
  <label
    htmlFor="title"
  >
     Title 
  </label>
  <input
    className="CakeForm-input"
    id="title"
    name="title"
    onChange={undefined}
    placeholder="Title"
    required={true}
    type="text"
    value={undefined}
  />
  <label
    htmlFor="desc"
  >
     Description 
  </label>
  <textarea
    className="CakeForm-input"
    cols="30"
    id="desc"
    name="desc"
    onChange={undefined}
    placeholder="Description"
    rows="10"
    value={undefined}
  />
  <label
    htmlFor="image"
  >
     Image 
  </label>
  <input
    className="CakeForm-input"
    id="image"
    name="image"
    onChange={undefined}
    placeholder="Image URL"
    type="url"
    value={undefined}
  />
  <button
    className="CakeForm-submit CakeForm-input"
    onClick={undefined}
    type="submit"
  >
    Add
  </button>
</form>
`;
